### PR TITLE
support multiple singing key types 

### PR DESCRIPTION
**Pull request by cburroughs**
Using the new support in java-http-signature 4.0.0, support DSA and
ECDSA keys.  A user can provide a key and the library will figure out
it's type and use sensible defaults.

As part of this I had to tease apart how `ThreadLocalSigner` objects
are managed by centralizing their creation in `MantaClient`.  It was
too easy to end up with multiple ones with different configuration
which only accidental worked when everything was RSA.  While several
class changed, the public api of `MantaClient` did not.

### CODE REVIEW

**Comment by cburroughs** [comment]
NOTE: That this depend on `SNAPSHOT`, so I'm not sure when you want to land this relative to the release.

**Comment by cburroughs** [comment]
I see one place in UriSigner that still has a hard coded RSA reference I need to figure out.

**Comment by cburroughs** [comment]
> I see one place in UriSigner that still has a hard coded RSA reference I need to figure out.

Okay I think I see the (small) change that has to happen, but it will require a change to http-sig to expose a string like "rsa-sha256".  I think I just never got to "signer" when I left the integration tests running for a while.  I'll should be able to get those diffs up this afternoon.

File: pom.xml [hunk 1]
```diff
@@ -252,6 +252,10 @@
                          only supports >=6 -->
                     <source>${java.min.version}</source>
                     <target>${java.min.version}</target>
+                    <compilerArgs>
```
Comment (dekobon) [inline pom.xml]: We might as well set: `-Xlint:all`

**Comment by cburroughs** [comment]
With the latest diff (and latests http-sig) I could run all ` -Dit.test=MantaClient*IT`, so I hope there are no more surprises.

**Comment by cburroughs** [comment]
(I think this is ready to land, but the hope was to get a release out of java-http-signature first)

**Comment by dekobon** [comment]
Yes, let's cut a version of the latest HTTP signatures and then I will run an integration test of this branch. If all goes well, then I will push it out as a snapshot.

**Pull request merged**